### ISSUE TITLE
4.12.1 release checklist

### ISSUE THREAD

**Issue by corinagum** [microsoft/BotFramework-WebChat#3789]
# Checklist

## Build

1. [x] Bump MockBot to Bot Framework SDK release 4.12.0
   - https://www.npmjs.com/package/botbuilder/v/4.12.0
   - (We will not bump for patch release)
1. [x] ~Bump `botframework-directlinejs` to `0.14.1`~
   - No newer version
1. [x] Bump to `4.12.1`
   - [x] Update `CHANGELOG.md` to mark specific changes in `4.12.1`
   - [x] Run `npm version --no-git-tag-version 4.12.1`
   - [x] Merged into `master`, the PR number is #3791
   - Commit is `7ec04e3`
   - Do not merge any other unrelated changes after this PR. Any other PR merged, will need to be re-tested
1. [x] Run daily pipeline manually, set "generate release version number" to `true`
   - (This will not push to NPM or CDN)
   - Pipeline name is `BotFramework-WebChat-daily`
   - The build number is [`222572`](https://fuselabs.visualstudio.com/BotFramework-WebChat/_build/results?buildId=222572&view=results) and commit is `7a90b6c`
 1. [x] Run WebChat-release-testing pipeline (should be automatic)
 1. [x] Add manual tests to WebChat-release-testing as needed

## Test

> The test should run against the build artifacts from Azure Pipelines.

1. [x] Manual testing on major browsers using `webchat-release-testing`
   - [x] Before starting testing, update all the browser version to latest
   - [x] Chrome 88.0.4324.190
   - [x] Edge 88.0.705.81
   - [x] Firefox 86.0
   - [x] IE11 (Windows 10 21322.1010)
   - [x] macOS Safari 13.1.3 (15609.4.1)
      - DLSpeech: Web Socket connection to `westus2.convai.speech.microsoft.com` closed instantly after connected, it also repro on previous versions of Web Chat (failed on 4.9.0, 4.8.1)
         - https://github.com/microsoft/cognitive-services-speech-sdk-js/issues/221
   - [x] iOS Safari 14.4
   - [x] Android Chrome 89.0.4389.86
1. [x] Test specific fixes related to `4.12.0` and previous releases
   
   - AC max parser
   - DLS `replyToId`

## Release
1. [x] Verify on WebChat-release-testing
1. [x] Make sure you are on `master` ~or `qfe`~ branch, run `git status` to check
1. [x] `git pull`
1. [x] Verify `/package.json`, `/package-lock.json`, and `CHANGELOG.md` has a version of `4.12.1`
1. [x] `git log`
   - Verify the latest commit is `7a90b6c`
1. [x] `git tag v4.12.1`
1. [x] `git push -u upstream v4.12.1`
   - You do not need to kick off a build again, use the previous build
1. [x] Create a new GitHub release, copy entries from `CHANGELOG.md`
   - [x] Subresource Integrity can be generated by
      - From CDN: `curl -H 'Accept-Encoding: gzip' https://cdn.botframework.com/botframework-webchat/4.12.0/webchat.js | gunzip - | openssl dgst -sha384 -binary | openssl base64 -A`
      - From local: `cat webchat.js | openssl dgst -sha384 -binary | openssl base64 -A`
   - [x] Attach assets including 3 JS files, `stats.json` and 5 tarballs
      - You can copy the artifacts from `webchat-release-testing/drops`
      - Tarballs download from npmjs
         ```
         curl -L -o botframework-directlinespeech-sdk-4.12.0.tgz https://registry.npmjs.org/botframework-directlinespeech-sdk/-/botframework-directlinespeech-sdk-4.12.0.tgz
         curl -L -o botframework-webchat-4.12.0.tgz https://registry.npmjs.org/botframework-webchat/-/botframework-webchat-4.12.0.tgz
         curl -L -o botframework-webchat-core-4.12.0.tgz https://registry.npmjs.org/botframework-webchat-core/-/botframework-webchat-core-4.12.0.tgz
         curl -L -o botframework-webchat-api-4.12.0.tgz https://registry.npmjs.org/botframework-webchat-api/-/botframework-webchat-api-4.12.0.tgz
         curl -L -o botframework-webchat-component-4.12.0.tgz https://registry.npmjs.org/botframework-webchat-component/-/botframework-webchat-component-4.12.0.tgz
         ```
1. [x] Kick off release to NPM
   - Release name is `[[PROD]]Push-WebChat-to-npmjs`
   - The build number is [222572](https://fuselabs.visualstudio.com/BotFramework-WebChat/_build/results?buildId=222572&view=results) release number is `31` and commit is `7a90b6c`
   - Retain the build indefinitely
1. [x] Kick off release to CDN (CDN CUTOFF TIME IS 2PM PST)
   1. [x] Prepare the email for approval
   - Release name is `[[PROD]]Push-WebChat-to-Prod-CDN-with-approval`
   - The build number is [`222572`](https://fuselabs.visualstudio.com/BotFramework-WebChat/_build/results?buildId=222572view=results), release number is [`36`](https://fuselabs.visualstudio.com/BotFramework-WebChat/_releaseProgress?_a=release-pipeline-progress&releaseId=2679) and commit is `7a90b6c`
   - Script build number is `131156` (this is fixed)
   1. [x] Send reminder email to approvers
   - Retain the build indefinitely

## Post-release check

- [x] Test using `webchat-release-testing`
   1. [x] Clone https://github.com/corinagum/WebChat-release-testing/
   1. [x] `01.create-react-app`
      1. [x] Nuke `01.create-react-app/node_modules`
      1. [x] `npm install`
      1. [x] `npm install botframework-webchat@4.12.1` (just install the bundle package)
      1. [x] `npm run build`
   1. [x] Others
      -  Using script tags from https://github.com/microsoft/BotFramework-WebChat/releases/tag/v4.12.1, with subresource integrity
         ```
         <script
            crossorigin="anonymous"
            integrity="sha384-"
            src="https://cdn.botframework.com/botframework-webchat/latest/webchat-es5.js"
         ></script>
         ```
   1. [x] `npx serve` (at repo root)
   1. [x] Go to http://localhost:5000/ to test, including IE11

## Notification to interested parties

- [x] Update partner page on Adaptive Cards doc
   - https://docs.microsoft.com/en-us/adaptive-cards/resources/partners
   - PR at https://github.com/MicrosoftDocs/AdaptiveCards/pull/370
- [ ] Notify related parties for the following fixes
   - [x] SDK team
   - [ ] Omnichannel
   - [x] Pooja
- [x] Update root README.md with feature notes -- Note: PR will be combined with post-release checklist PR
- [ ] Update https://github.com/microsoft/BotFramework-sdk/issues/6200
